playwright/pages/dialogs/addPromotionRuleDialog.ts

```diff
@@ -0,0 +1,70 @@
+        readonly selectOption = page.getByTestId(
+            "select-option",
+        ),
+        readonly ruleDescriptionInput = page.getByTestId("rich-text-editor-description"),
+        readonly addRuleConditionButton = page.getByTestId("add-rule-condition-button"),
+        readonly addRuleConditionSection = page.getByTestId("conditions-section"),
+        readonly addRuleConditionNameInput = page.getByTestId("condition-name-0"),
+        readonly addRuleConditionValueInput = page.getByTestId("condition-value-0"),
+        readonly addRuleConditionTypeDropdown = page.getByTestId("condition-type-0"),
+        readonly conditionRemoveButton = page.getByTestId("condition-remove-0"),
+        readonly rewardTypeSelect = page.getByTestId("reward-type-select"),
+        readonly percentageRewardValueTypeOption = page.getByTestId("percentage-reward-value-type"),
+        readonly fixedRewardValueTypeOption = page.getByTestId("fixed-reward-value-type"),
+        readonly rewardGiftSelect = page.getByTestId("reward-gifts-select"),
+    ) {
+        this.page = page;
+    }
+    async typePromotionRuleDescription(description: string) {
+        await this.ruleDescriptionInput.fill(description);
+    }
+    async clickChannelsDropdown() {
+        await this.ruleChannelDropdown.click();
+    }
+    async selectSingleChannel(channel = "PLN") {
+        await this.selectOption.filter({ hasText: `Channel-${channel}` }).click();
+        // below click hides prompted options
+        this.clickChannelsDropdown();
+        await this.page.getByRole('option', { 'selected': true }).waitFor({ state: 'visible' });
+        await this.addRuleConditionButton.waitFor({
+            state: "visible",
+            timeout: 10000,
+        });
+    }
+    async typePromotionRuleName(name: string) {
+        await this.ruleNameInput.fill(name);
+    }
+    async clickAddRuleConditionButton() {
+        await this.addRuleConditionButton.click();
+        await this.addRuleConditionSection.waitFor({
+            state: "visible",
+            timeout: 10000,
+        });
+    }
+    async selectPercentageRewardValueType() {
+        await this.percentageRewardValueTypeOption.click();
+        // await this.percentageRewardValueTypeOption.getByRole('checkbox', { 'checked': true }).waitFor({ state: 'visible' });
+        await expect(this.percentageRewardValueTypeOption).toBeChecked();}
+    async selectFixedRewardValueType() {
+        await this.fixedRewardValueTypeOption.click();
+        await expect(this.fixedRewardValueTypeOption).toBeChecked();
+    }
+    async selectRewardType(type: string) {
+        await this.rewardTypeSelect.click();
+        await this.page.getByRole("listbox").waitFor({
+            state: "visible",
+            timeout: 10000,
+        });
+        // await this.page.getByTestId("select-option").getByText(type).click();
+        await this.page.getByTestId("select-option").selectOption(type);
```

Comment:
question: why not use `this.selectOption` here?

Reply:
@poulch I've removed `addPromotionRuleDialog.ts` file for now as it is not needed until I create tests for promotion rules (will be added back in the next PR).